test/flux_calculator_tests.jl

```diff
@@ -50,3 +63,222 @@ end
+    sim::TestOcean,
+    thermo_params::TD.Parameters.ThermodynamicsParameters,
+    thermo_state_int,
+    colidx::Fields.ColumnIndex,
+)
+    T_sfc = Interfacer.get_field(sim, Val(:surface_temperature), colidx)
+    ρ_sfc = thermo_state_int.ρ # arbitrary
+    q_sfc = Interfacer.get_field(sim, Val(:air_humidity), colidx) # read from cache
+    @. TD.PhaseEquil_ρTq.(thermo_params, ρ_sfc, T_sfc, q_sfc)
+end
+
+function update_turbulent_fluxes_point!(sim::TestOcean, fields::NamedTuple, colidx::Fields.ColumnIndex)
+    (; F_turb_energy) = fields
+    (; F_turb_energy) = fields
+    @. sim.integrator.p.F_aero[colidx] = F_turb_energy
+end
+
+@testset "calculate correct fluxes: dry" begin
+    surface_scheme_list = (MoninObukhovScheme(), BulkScheme())
+    for scheme in surface_scheme_list
+        boundary_space = TestHelper.create_space(FT)
+
+        params = (; surface_scheme = scheme, FT = FT)
+
+        # atmos
+        p = (;
+            energy_bc = zeros(boundary_space),
+            ρq_tot_bc = zeros(boundary_space),
+            uₕ_bc = ones(boundary_space),
+            z = ones(boundary_space),
+            z_sfc = zeros(boundary_space),
+            u = ones(boundary_space),
+            v = ones(boundary_space),
+        )
+        Y_init = (; ρ = ones(boundary_space) .* 1.2, T = ones(boundary_space) .* 310, q = zeros(boundary_space))
+        integrator = (; Y_init..., p = p)
+        atmos_sim = TestAtmos(params, Y_init, nothing, integrator)
+
+        # ocean
+        p = (;
+            F_aero = zeros(boundary_space),
+            z0m = FT(0.01),
+            z0b = FT(0.01),
+            beta = ones(boundary_space),
+            α = ones(boundary_space) .* 0.5,
+            q = zeros(boundary_space),
+            Cd = FT(0.01),
+            Ch = FT(0.01),
+            area_fraction = ones(boundary_space) .* 0.5,
+        )
+        Y_init = (; T = ones(boundary_space) .* 300.0)
+        integrator = (; Y_init..., p = p)
+        ocean_sim = TestOcean(nothing, Y_init, nothing, integrator)
+
+        # ocean
+        ocean_sim2 = TestOcean(nothing, Y_init, nothing, integrator)
+
+        model_sims = (; atmos_sim, ocean_sim, ocean_sim2)
+
+        coupler_cache_names = (
+            :T_S,
+            :albedo,
+            :F_R_sfc,
+            :F_R_toa,
+            :P_liq,
+            :P_snow,
+            :P_net,
+            :F_turb_energy,
+            :F_turb_ρτxz,
+            :F_turb_ρτyz,
+            :F_turb_moisture,
+        )
+        fields = NamedTuple{coupler_cache_names}(ntuple(i -> Fields.zeros(boundary_space), length(coupler_cache_names)))
+
+        # calculate turbulent fluxes
+        thermo_params = get_thermo_params(atmos_sim)
+        partitioned_turbulent_fluxes!(model_sims, fields, boundary_space, scheme, thermo_params)
+
+        # calculating the fluxes twice ensures that no accumulation occurred
+        partitioned_turbulent_fluxes!(model_sims, fields, boundary_space, scheme, thermo_params)
```

Comment:
I'm not sure I follow the need for a double call here. Do you mean accumulation in time? That should be frozen at the time of calculating the fluxes

Reply:
I also don't totally understand - could you add more comments please? Also maybe note that this is something we're planning to change

Reply:
yes, the double call ensures that the result is deterministic based on the current state (i.e. the fluxes are reset every time this is called). And sure, I can add more comments. Thanks both!